src/components/chips/js/chipController.js

```diff
@@ -0,0 +1,165 @@
+angular
+    .module('material.components.chips')
+    .controller('MdChipCtrl', MdChipCtrl);
+
+/**
+ * Controller for the MdChip component. Responsible for handling keyboard
+ * events and editting the chip if needed.
+ *
+ * @param $scope
+ * @param $element
+ * @param $mdConstant
+ * @param $timeout
+ * @constructor
+ */
+function MdChipCtrl ($scope, $element, $mdConstant, $attrs, $timeout) {
+    /**
+     * @type {$element}
+     */
+    this.$element = $element;
+
+    /**
+     * @type {$mdConstant}
+     */
+    this.$mdConstant = $mdConstant;
+
+    /**
+     * @type {$timeout}
+     */
+    this.$timeout = $timeout;
+
+    /**
+     * @type {boolean}
+     */
+    this.isEditting = false;
+
+    /**
+     * @type {boolean}
+     */
+    this.enableChipEdit = $scope.$parent && $scope.$parent.enableChipEdit == 'true';
+
+    if(this.enableChipEdit) {
+        this.$element.on('keydown', this.chipKeyDown.bind(this));
```

Comment:
You should indent your code with two spaces to be aligned with the other components.